users/pcoves/rainbowUnicorn.c

```diff
@@ -0,0 +1,42 @@
+#include "rainbowUnicorn.h"
+#include "pcoves.h"
+
+static struct {
+    bool    enabled;
+    uint8_t color;
+    char    string[2];
+    uint8_t mods;
+} state = {false, 0};
+
+bool process_record_rainbowUnicorn(uint16_t keycode, keyrecord_t* record) {
+    if (keycode == RAINBOW_UNICORN_TOGGLE) {
+        state.enabled ^= true;
+        return false;
+    }
+
+    if (!state.enabled) return true;
+
+    switch (keycode) {
+        case KC_A ... KC_Z:
+        case KC_1 ... KC_0:
+        case ALT_T(KC_A)... ALT_T(KC_Z):
+        case CTL_T(KC_A)... CTL_T(KC_Z):
+        case GUI_T(KC_A)... GUI_T(KC_Z):
+        case SFT_T(KC_A)... SFT_T(KC_Z):
```

Comment:
Can you please describe what is the effect of this rainbow unicorn, how do you use it?
Came here from https://www.reddit.com/r/ErgoMechKeyboards/comments/ek0418/rainbow_gergo/fd5b6e4/

Reply:
Hi there.

The `rainbow unicorn` feature is the result of me not wanting to add a plugin to `irssi`.

On *IRC*, one can indicate a color for both the characters and their background.
This piece of code, one enabled will prefix every letter you type by a color code making your text looking like a rainbow, hence `rainbow unicorn`, cause... you know... **unicorns**!

In order to use it (but do you really want to use it?) you have to:
* Enable it in the firmware by setting `UNICODE_ENABLE` to `true` https://github.com/qmk/qmk_firmware/blob/master/users/pcoves/pcoves.h#L9 .
* Add the `RAINBOW_UNICORN_TOGGLE` keycode to your keymap https://github.com/qmk/qmk_firmware/blob/master/users/pcoves/pcoves.h#L10 .

It should be enough and acts as a switch so press it again to disable.

Again, this is just a nonsensical piece of code I wrote out of boredom, a software feature in your IRC client would be a better candidate for such a feature.

Reply:
Ahaha. Sounds pretty useful, even for such a niche use case. A nice example when you can put more logic in that small computer with keys that connects to your bigger computer.
Learned a couple of tricks from your userspace, e.g. this `static struct` for keeping some state. Might be of some use when I come to rework the default layout on my kb.
Thanks!

Reply:
@drashna deserves most of your thanks as my userspace is heavily inspired by his own. You might find other useful stuff there.

#enjoy